queries.js

```diff
@@ -0,0 +1,171 @@
+const dayjs = require('dayjs');
+
+module.exports = (self, query) => {
+  return {
+    builders: {
+      future: {
+        def: null,
+        async finalize() {
+          const future = query.get('future');
+          if (future === null || query.get('year') || query.get('month')) {
+            return;
+          }
+
+          const today = dayjs().format('YYYY-MM-DD');
+          if (future) {
+            query.and({ releaseDate: { $gte: today } });
+          } else {
+            query.and({ releaseDate: { $lte: today } });
+          }
+        },
+        launder(value) {
+          return self.apos.launder.booleanOrNull(value);
+        },
+        choices() {
+          return [
+            {
+              value: true,
+              label: 'aposBlog:future'
+            },
+            {
+              value: false,
+              label: 'aposBlog:past'
+            },
+            {
+              value: null,
+              label: 'aposBlog:both'
+            }
+          ];
+        }
+      },
+
+      // Filter by year, in YYYY format.
+      year: {
+        def: null,
+        safeFor: 'public',
+        finalize() {
+          const year = query.get('year');
+          if (!year) {
+            return;
+          }
+
+          query.and({ releaseDate: { $regex: '^' + year } });
+        },
+        launder(value) {
+          return self.apos.launder.string(value);
```

Comment:
Must launder that it actually matches four digits.

Reply:
Fixed